### ISSUE TITLE
MessageChannel constructor performance

### ISSUE THREAD

**Issue by davidchappelle** [nodejs/help#3025]
* **Node.js Version**: v12.18.4
* **OS**: Ubuntu 18.04 LTS
* **Scope (install, code, runtime, meta, other?)**: code/runtime
* **Module (and version) (if relevant)**: worker_threads

I am using perf_hooks to help track down some issues that I have with the event loop getting blocked. I am measuring the creation of MessageChannel objects and noticed that sometimes it can take upwards of 20ms to create a MessageChannel object. I am curious where this latency is coming from as it seems that the creation of a MessageChannel object should be more or less constant time and very lightweight:
```js
            const start = performance.now();
            channel = new MessageChannel();
            const end = performance.now();
            console.log(`MessageChannel:constructor ${end - start} ms`);
```

I also tried to find the underlying code for the MessageChannel constructor but was unable to locate any specific code. If someone could point me to that code as well that would be great.

**Comment by addaleax** [comment]
> I also tried to find the underlying code for the MessageChannel constructor but was unable to locate any specific code.

That would be here: 

https://github.com/nodejs/node/blob/ab0af50ef8cd18eff18918cf380ad1127d4c4a24/src/node_messaging.cc#L1285-L1301

here:

https://github.com/nodejs/node/blob/ab0af50ef8cd18eff18918cf380ad1127d4c4a24/src/node_messaging.cc#L647-L680

here:

https://github.com/nodejs/node/blob/ab0af50ef8cd18eff18918cf380ad1127d4c4a24/src/node_messaging.cc#L577-L615

and here:

https://github.com/nodejs/node/blob/ab0af50ef8cd18eff18918cf380ad1127d4c4a24/lib/internal/worker/io.js#L142-L150

In particular the fact that this calls from JS into C++ and back into JS might take some time, but it’s hard to tell what exactly is causing the trouble in your script, since running JS code only once will generally not run the optimized version of it.

**Comment by davidchappelle** [comment]
Thanks for the quick response @addaleax . Although my example code above only shows a single invocation, the issue I am describing seems to happen randomly over time. I will put together a more complete test case that illustrates the issue. I opened a similar help issue for postMessage #3026 , every 20-30 invocations I get a `postMessage` call that blocks the event loop anywhere from say 10 to 70 or so milliseconds.

**Comment by PoojaDurgad** [comment]
duplicate of https://github.com/nodejs/help/issues/3026
